### ISSUE TITLE
Play does not display relevant Hibernate message on JDK12

### ISSUE THREAD

**Issue by eximius313** [playframework/playframework#9171]
### Play Version (2.5.x / etc)
2.6.21

### API (Scala / Java / Neither / Both)
N/A

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
Windows 7

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
```
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)
```

### Expected Behavior
When I run exactly the same project on JDK 9.0.4 I get relevant message:
```
play.api.UnexpectedException: Unexpected exception[CreationException: Unable to create injector, see the following errors:

1) Error injecting constructor, javax.persistence.PersistenceException: [PersistenceUnit: defaultPersistenceUnit] Unable to build Hibernate SessionFactory
  at play.db.jpa.DefaultJPAApi$JPAApiProvider.<init>(DefaultJPAApi.java:44)
  at play.db.jpa.DefaultJPAApi$JPAApiProvider.class(DefaultJPAApi.java:39)
  while locating play.db.jpa.DefaultJPAApi$JPAApiProvider
  while locating play.db.jpa.JPAApi
    for the 1st parameter of DynamicHandler.<init>(DynamicHandler.java:149)
  while locating DynamicHandler
    for field at SecurityHandler.dynamicHandler(SecurityHandler.java:53)
  while locating SecurityHandler
  at lib.ApplicationModule.configure(ApplicationModule.java:135) (via modules: com.google.inject.util.Modules$OverrideModule -> lib.ApplicationModule)
  while locating be.objectify.deadbolt.java.DeadboltHandler annotated with @HandlerQualifiers$MainHandler()
    for the 1st parameter of SecurityHandlerCache.<init>(SecurityHandlerCache.java:21)
  at SecurityHandlerCache.class(SecurityHandlerCache.java:12)
  while locating SecurityHandlerCache
  at lib.ApplicationModule.configure(ApplicationModule.java:138) (via modules: com.google.inject.util.Modules$OverrideModule -> lib.ApplicationModule)
  while locating be.objectify.deadbolt.java.cache.HandlerCache

1 error]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:186)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:124)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:241)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:138)
        at akka.stream.impl.fusing.MapAsyncUnordered$$anon$31.onPush(Ops.scala:1318)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:588)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:472)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:745)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:760)
        at akka.actor.Actor.aroundReceive(Actor.scala:517)
        at akka.actor.Actor.aroundReceive$(Actor.scala:515)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:670)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
        at akka.actor.ActorCell.invoke(ActorCell.scala:557)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
        at akka.dispatch.Mailbox.run(Mailbox.scala:225)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Error injecting constructor, javax.persistence.PersistenceException: [PersistenceUnit: defaultPersistenceUnit] Unable to build Hibernate SessionFactory
  at play.db.jpa.DefaultJPAApi$JPAApiProvider.<init>(DefaultJPAApi.java:44)
  at play.db.jpa.DefaultJPAApi$JPAApiProvider.class(DefaultJPAApi.java:39)
  while locating play.db.jpa.DefaultJPAApi$JPAApiProvider
  while locating play.db.jpa.JPAApi
    for the 1st parameter of DynamicHandler.<init>(DynamicHandler.java:149)
  while locating DynamicHandler
    for field at SecurityHandler.dynamicHandler(SecurityHandler.java:53)
  while locating SecurityHandler
  at lib.ApplicationModule.configure(ApplicationModule.java:135) (via modules: com.google.inject.util.Modules$OverrideModule -> lib.ApplicationModule)
  while locating be.objectify.deadbolt.java.DeadboltHandler annotated with @HandlerQualifiers$MainHandler()
    for the 1st parameter of SecurityHandlerCache.<init>(SecurityHandlerCache.java:21)
  at SecurityHandlerCache.class(SecurityHandlerCache.java:12)
  while locating SecurityHandlerCache
  at lib.ApplicationModule.configure(ApplicationModule.java:138) (via modules: com.google.inject.util.Modules$OverrideModule -> lib.ApplicationModule)
  while locating be.objectify.deadbolt.java.cache.HandlerCache

1 error
        at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:543)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:186)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
        at com.google.inject.Guice.createInjector(Guice.java:87)
        at com.google.inject.Guice.createInjector(Guice.java:78)
        at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:185)
        at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:137)
        at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
        at play.core.server.DevServerStart$$anon$1.$anonfun$reload$3(DevServerStart.scala:174)
        at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:171)
        ... 24 common frames omitted
Caused by: javax.persistence.PersistenceException: [PersistenceUnit: defaultPersistenceUnit] Unable to build Hibernate SessionFactory
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.persistenceException(EntityManagerFactoryBuilderImpl.java:1015)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:941)
        at org.hibernate.jpa.HibernatePersistenceProvider.createEntityManagerFactory(HibernatePersistenceProvider.java:56)
        at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
        at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:39)
        at play.db.jpa.DefaultJPAApi.lambda$start$0(DefaultJPAApi.java:65)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at play.db.jpa.DefaultJPAApi.start(DefaultJPAApi.java:64)
        at play.db.jpa.DefaultJPAApi$JPAApiProvider.<init>(DefaultJPAApi.java:51)
        at play.db.jpa.DefaultJPAApi$JPAApiProvider$$FastClassByGuice$$dcd4cdbd.newInstance(<generated>)
        at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:60)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:52)
        at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:147)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:124)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:211)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:182)
        ... 33 common frames omitted
Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: missing column [my_id] in table [my_table]
        at org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateTable(AbstractSchemaValidator.java:136)
        at org.hibernate.tool.schema.internal.GroupedSchemaValidatorImpl.validateTables(GroupedSchemaValidatorImpl.java:42)
        at org.hibernate.tool.schema.internal.AbstractSchemaValidator.performValidation(AbstractSchemaValidator.java:89)
        at org.hibernate.tool.schema.internal.AbstractSchemaValidator.doValidation(AbstractSchemaValidator.java:68)
        at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:191)
        at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72)
        at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:309)
        at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:938)
        ... 77 common frames omitted
```

### Actual Behavior
Instead on JDK 12 I get:
```
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3962)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3966)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4951)
        at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4957)
        at com.google.inject.internal.util.StackTraceElements.forMember(StackTraceElements.java:71)
        at com.google.inject.internal.Messages.formatParameter(Messages.java:282)
        at com.google.inject.internal.Messages.formatInjectionPoint(Messages.java:273)
        at com.google.inject.internal.Messages.formatSource(Messages.java:229)
        at com.google.inject.internal.Messages.formatSource(Messages.java:220)
        at com.google.inject.internal.Messages.formatMessages(Messages.java:90)
        at com.google.inject.CreationException.getMessage(CreationException.java:50)
        at play.api.UnexpectedException$$anonfun$$lessinit$greater$1.$anonfun$new$1(Exceptions.scala:12)
        at scala.Option.map(Option.scala:146)
        at play.api.UnexpectedException$$anonfun$$lessinit$greater$1.apply(Exceptions.scala:12)
        at play.api.UnexpectedException$$anonfun$$lessinit$greater$1.apply(Exceptions.scala:12)
        at scala.Option.getOrElse(Option.scala:121)
        at play.api.UnexpectedException.<init>(Exceptions.scala:12)
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:186)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:124)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:241)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:138)
        at akka.stream.impl.fusing.MapAsyncUnordered$$anon$31.onPush(Ops.scala:1318)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:588)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:472)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:745)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:760)
        at akka.actor.Actor.aroundReceive(Actor.scala:517)
        at akka.actor.Actor.aroundReceive$(Actor.scala:515)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:670)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
        at akka.actor.ActorCell.invoke(ActorCell.scala:557)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
        at akka.dispatch.Mailbox.run(Mailbox.scala:225)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.IllegalArgumentException: null
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:160)
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:143)
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:418)
        at com.google.inject.internal.util.LineNumbers.<init>(LineNumbers.java:64)
        at com.google.inject.internal.util.StackTraceElements$1.load(StackTraceElements.java:49)
        at com.google.inject.internal.util.StackTraceElements$1.load(StackTraceElements.java:45)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3523)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2249)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2132)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045)
```

### Reproducible Test Case
Just run Play with inconsistent database state on JDK12

**Comment by mkurz** [comment]
Play officially only supports Java 8 right now (and I guess 2.7.x should work with Java 11 as well, however I would wait for Scala 2.13 support because AFAIK it comes with better support for JDK 11).
Also are you sure that the Hibernate version you use already officially supports Java 12?

If I were you I would not run any Play production applications on Java versions which are not LTS because you will run into many problems e.g. Java 12 is supported only for 6 month, and if you don't upgrade to Java 13 later (because you can't because of issues like this) you are stuck with an unpatched JVM if security bugs arise.
It's safer to stay on Java 8 or 11 IMHO.

**Comment by eximius313** [comment]
@mkurz JDK11 in it's current OpenJDK version (11.0.2) is completly unusable with `Hibrnate metamodel generator` because of this bug: https://bugs.openjdk.java.net/browse/JDK-8220578
I'll migrate to 11.0.4 as soon as it is released 

When do you plan to support Scala 2.13?

**Comment by mkurz** [comment]
@eximius313 Play 2.7.0 was published for Scala 2.13.0-M5 already, so I think you should be able to use that version until the final release approaches.

Scala 2.13.0-RC1 should be released today or tomorrow, see https://contributors.scala-lang.org/t/2-13-0-rc1-timing-january-2019/2375/9
Afterwards it will take another couple of weeks I guess until 2.13.0 final will be released and it will also take a while until all projects and libraries will be published for it.

**Comment by mkurz** [comment]
This will get fixed by #10724 - Guice 5 supports class file versions for Java 17 already.

The origin of the error in this issue here is Guice (or better the ASM version Guice is using):
```
Caused by: java.lang.IllegalArgumentException: null
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:160)
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:143)
        at com.google.inject.internal.asm.$ClassReader.<init>(ClassReader.java:418)
        at com.google.inject.internal.util.LineNumbers.<init>(LineNumbers.java:64)
        at com.google.inject.internal.util.StackTraceElements$1.load(StackTraceElements.java:49)
        at com.google.inject.internal.util.StackTraceElements$1.load(StackTraceElements.java:45)
```
